honeycomb-core/src/twomap.rs

```diff
@@ -922,45 +366,174 @@ impl<T: CoordsFloat> CMap2<T> {
     /// - `betas: [DartIdentifier; 3]` -- Value of the images as
     ///   *[β<sub>0</sub>(dart), β<sub>1</sub>(dart), β<sub>2</sub>(dart)]*
     ///
-    /// # Example
-    ///
-    /// See [CMap2] example.
-    ///
     pub fn set_betas(&mut self, dart_id: DartIdentifier, betas: [DartIdentifier; CMAP2_BETA]) {
         self.betas[dart_id as usize] = betas;
     }
+}
 
-    /// Set the vertex ID associated to a dart.
+// --- icell-related code
+impl<T: CoordsFloat> CMap2<T> {
+    /// Fetch vertex identifier associated to a given dart.
     ///
     /// # Arguments
     ///
-    /// - `dart_id: DartIdentifier` -- ID of the dart of interest.
-    /// - `vertex_id: VertexIdentifier` -- Unique vertex identifier.
+    /// - `dart_id: DartIdentifier` -- Identifier of *dart*.
     ///
-    /// # Example
+    /// # Return / Panic
     ///
-    /// See [CMap2] example.
+    /// Return the identifier of the associated vertex.
     ///
-    pub fn set_vertexid(&mut self, dart_id: DartIdentifier, vertex_id: VertexIdentifier) {
-        self.dart_data.associated_cells[dart_id as usize].vertex_id = vertex_id;
+    pub fn vertex_id(&self, dart_id: DartIdentifier) -> VertexIdentifier {
+        Orbit2::new(self, OrbitPolicy::Vertex, dart_id)
+            .min()
+            .unwrap() as VertexIdentifier
     }
```

Comment:
Essentially, an orbit is always non-empty, therefore the `min()` method will always yield a result. The fact that the orbit is always non-empty is its definition: it will always include the dart passed as an argument since it is the starting point of the internal BFS. Because of that, all darts (even the null dart!) yield a non-empty orbit (given that the policy is valid).